### PR TITLE
Minor changes for DB for MySQL-Flex Server TLS Control

### PR DESCRIPTION
**Pull request by HarshitArora8**

### CODE REVIEW

File: Scripts/RemediationScripts/Remediate-SetDBForMySQLFlexibleServerMinReqTLSVersion.ps1 [hunk 1]
```diff
@@ -334,9 +334,9 @@ function Set-SecureTLSVersionForDBForMySQLFlexibleServer {
                     $DBForMySQLFSResource = Get-AzMySqlFlexibleServer -ResourceGroupName $_.ResourceGroupName -Name $_.ResourceName -ErrorAction SilentlyContinue
             
                     $parameterValue = (Get-AzMySqlFlexibleServerConfiguration -Name $([Constants]::ParameterName) -ResourceGroupName $_.ResourceGroupName -ServerName $_.ResourceName -SubscriptionId $SubscriptionId).Value
-                    $DBForMySQLFlexibleServerDetails += $_  | Select-Object @{N = 'ResourceId'; E = { $_.Id } },
-                    @{N = 'ResourceGroupName'; E = { $_.Id.Split("/")[4] } },
-                    @{N = 'ResourceName'; E = { $_.Name } }, 
+                    $DBForMySQLFlexibleServerDetails += $_  | Select-Object @{N = 'ResourceId'; E = { $_.ResourceId } },
+                    @{N = 'ResourceGroupName'; E = { $_.ResourceGroupName } },
```
Comment (asimks) [inline Scripts/RemediationScripts/Remediate-SetDBForMySQLFlexibleServerMinReqTLSVersion.ps1:338]: Same changes should be done to the auto remediation mode code flow to keep things uniform.

Comment (HarshitArora8) [reply Scripts/RemediationScripts/Remediate-SetDBForMySQLFlexibleServerMinReqTLSVersion.ps1]: I have done the same change as we are doing with auto remediation and from the subscription as well. Using ID and splitting it to get the RG name now in both the condition.

File: Scripts/RemediationScripts/Remediate-SetDBForMySQLFlexibleServerMinReqTLSVersion.ps1 [hunk 2]
```diff
@@ -375,7 +374,28 @@ function Set-SecureTLSVersionForDBForMySQLFlexibleServer {
 
     if ($totalDBForMySQLFSWithNonSecureTLSVersionEnabled -eq 0) {
         Write-Host "No Azure Database for MySQL flexible server(s) found with non-secure TLS version enabled.. Exiting..." -ForegroundColor $([Constants]::MessageType.Warning)
-        Write-Host $([Constants]::DoubleDashLine)	
+        Write-Host $([Constants]::DoubleDashLine)
+        
+        if($AutoRemediation) 
+        {
+            $DBForMySQLFlexibleServerDetails | ForEach-Object {
```
Comment (saurabhgautam-msft) [inline Scripts/RemediationScripts/Remediate-SetDBForMySQLFlexibleServerMinReqTLSVersion.ps1:381]: No need to this.

File: Scripts/RemediationScripts/Remediate-SetDBForMySQLFlexibleServerMinReqTLSVersion.ps1 [hunk 3]
```diff
@@ -375,7 +374,28 @@ function Set-SecureTLSVersionForDBForMySQLFlexibleServer {
 
     if ($totalDBForMySQLFSWithNonSecureTLSVersionEnabled -eq 0) {
         Write-Host "No Azure Database for MySQL flexible server(s) found with non-secure TLS version enabled.. Exiting..." -ForegroundColor $([Constants]::MessageType.Warning)
-        Write-Host $([Constants]::DoubleDashLine)
```
Comment (saurabhgautam-msft) [inline Scripts/RemediationScripts/Remediate-SetDBForMySQLFlexibleServerMinReqTLSVersion.ps1:378]: Resolve merge conflicts.

File: Scripts/RemediationScripts/Remediate-SetDBForMySQLFlexibleServerMinReqTLSVersion.ps1 (hunk 2 above)
Comment (HarshitArora8) [reply Scripts/RemediationScripts/Remediate-SetDBForMySQLFlexibleServerMinReqTLSVersion.ps1]: Removed and logged skipped resources where we are checking the tls version.

File: Scripts/RemediationScripts/Remediate-SetDBForMySQLFlexibleServerMinReqTLSVersion.ps1 (hunk 3 above)
Comment (HarshitArora8) [reply Scripts/RemediationScripts/Remediate-SetDBForMySQLFlexibleServerMinReqTLSVersion.ps1:378]: Fixed.

**Review by saurabhgautam-msft — APPROVED** [review]
LGTM

**Review by asimks — APPROVED** [review]

**Pull request merged**